### PR TITLE
move to docker-compose v2

### PR DESCRIPTION
**Pull request by woodsaj**
- uses a docker-network which provides an embeded DNS server for
  host resolution. (no need to define "links" anymore unless you
  need aliases)
- update to use metrictank docker image from dockerHub
- update to use graphite-metrictank image from dockerhub
- update to use worldping-api image from dockerhub
- updates proxy to proxy both https://tsdb-gw and https://worldping-api

### CODE REVIEW

File: docker/worldping-api/worldping-api.ini [hunk 1]
```diff
@@ -20,7 +20,7 @@ cert_key =
 admin_key = changeme
 
 [raintank]
-graphite_url = http://graphite-api:8888/
+graphite_url = http://metrictank:6063/
```
Comment (Dieterbe) [inline docker/worldping-api/worldping-api.ini]: how does worldping-api use graphite_url ? shouldn't this be tsdb-gw ?

**Comment by Dieterbe** [comment]
> updates proxy to proxy both https://tsdb-gw and https://worldping-api

why do we need to proxy those?

File: docker/metrictank/metrictank.ini [hunk 2]
```diff
@@ -93,7 +83,7 @@ proftrigger-heap-thresh = 10000000
 log-min-dur = 5min
 
 # only log log-level and higher. 0=TRACE|1=DEBUG|2=INFO|3=WARN|4=ERROR|5=CRITICAL|6=FATAL
-log-level = 2
+log-level = 1
```
Comment (Dieterbe) [inline docker/metrictank/metrictank.ini]: why this change? i find debug to be overwhelming 99% of the time i run this stack.

File: docker/metrictank/metrictank.ini [hunk 3]
```diff
@@ -170,3 +160,18 @@ consumer-opt =
...
 
+
+[elasticsearch-idx]
+enabled = true
+# elasticsearch index name to use
+index = metric
+# Elasticsearch host addresses (multiple hosts can be specified as comma-separated list)
+hosts = elasticsearch:9200
+# how often the retry buffer should be flushed to ES
+retry-interval = 1h
```
Comment (Dieterbe) [inline docker/metrictank/metrictank.ini]: this is too long especially for short lived dev envs. let's stick to the default of 10m, or frankly even shorter

**Comment by Dieterbe** [comment]
> uses a docker-network which provides an embeded DNS server for host resolution. (no need to define "links" anymore unless you need aliases)

this sounds neat.

File: docker/worldping-api/worldping-api.ini (hunk 1 above)
Comment (woodsaj) [reply docker/worldping-api/worldping-api.ini]: legacy.  older versions of the worldping-app still point to worldping-api.raintank.io/api/graphite for querying the TSDB.

**Comment by woodsaj** [comment]
> why do we need to proxy those?

For the worldping-app.

File: docker/worldping-api/worldping-api.ini (hunk 1 above)
Comment (Dieterbe) [reply docker/worldping-api/worldping-api.ini]: ok but so it expects a graphite server and we're giving it the metrictank http port, which doesn't really have a graphite http api. that's what confuses me.

Comment (woodsaj) [reply docker/worldping-api/worldping-api.ini]: Well, the graphite server is used for both serving requests for worldping-app and also for alerting queries.
I had this pointed to MT for for testing, but your right it should point to to graphite-metrictank
